### DISCUSSION TITLE
[OP Stack Deployment] ops-bedrock with consensus node

### DISCUSSION THREAD

**Discussion by qu0b** [ethereum-optimism/developers#23] [Deploying the OP Stack] [answered]
### Issue Description

I was wondering why the ops-bedrock setup used in the Makefile has a l1 node with mining enabled instead of using a consensus node to stand up the optimism stack. It should be pretty trivial to add a prysm node in interop mode.

**Comment by opfocus** [comment]
> While the main network uses proof-of-stake (PoS) to secure the blockchain, Geth also supports the the 'Clique' proof-of-authority (PoA) consensus algorithm and the Ethash proof-of-work algorithm as alternatives for private networks. Clique is strongly recommended for private testnets because PoA is far less resource-intensive than PoW. The key differences between the consensus algorithms available in Geth are:

The information above is from the Geth documentation. Please forgive me for not knowing much; I'm just here to participate in the discussion. It seems that in the quoted text, there is no recommendation to use PoS but rather recommends PoA. 
I'd like to tag  @ajsutton here; he might be able to help us.

**Reply by opfocus** [reply]
It should be noted that the L1 mentioned is only used in devnet, not in op-stack

**Reply by sbvegan** [reply]
Ideally the devnet reflects the real world production version. I believe this documentation is for private networks. We wouldn't want to implement PoA because it goes further away from the real world implementation.

**Comment by sbvegan** [comment]
Its something that we still need to do. I just spoke to a client engineer and they said they're open to this type of contribution. This will be a requirement for when EIP-4844 lands, so it will have to be implemented by then.

**Reply by opfocus** [reply]
I see

**Reply by qu0b** [reply]
I've got something working along these lines, still cleaning it up. If you are interested I can prepare a PR for review and look into aligning it with the current status quo

**Reply by sbvegan** [reply, marked as answer]
Yeah it would definitely be appreciated, we'll want to direct the work to the monorepo. Here's the [contributing doc](https://github.com/ethereum-optimism/optimism/blob/develop/CONTRIBUTING.md#workflow-for-pull-requests). 

cc @tynes this one is coming your way :)

**Reply by sbvegan** [reply]
@qu0b as this migrates to the monorepo, can you mark this thread as answered so I can close the loop here?

**Reply by mtgnoah** [reply]
Here is one way to use PoS instead of the current solution. [https://github.com/OffchainLabs/eth-pos-devnet](ethposdevnet)
